### PR TITLE
feat(devops): Manually deploy breaking changes via CI

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
If an API changes, dfx will typically ask the user whether they really want to deploy.  In CI, we need a way of saying "yes", but probably don't want this to be the default.

There is an existing checkbox for forcing the backend deployment to take place, no matter what.  It seems reasonable to use `--yes` when that is selected, so we do not need a new input.

# Changes
- If `force-backend` is selected in the CI deployment task, deploy with `--yes` and `--upgrade-unchanged`.

# Tests
Executed the action by:
- Disabling the check that allows deploying only from main.
- Running the action with: `gh workflow run deploy-to-environment.yml --ref "yes" --field network=staging --field canister=backend --field force-backend=true`
- Re-enabling the main check.

### CODE REVIEW

File: .github/workflows/deploy-to-environment.yml [hunk 1]
```diff
@@ -45,7 +45,7 @@ jobs:
         if: ${{ github.ref != 'refs/heads/main' }}
         run: |
           echo "This workflow can only be manually triggered with workflow_dispatch on the main branch"
-          exit 1
+          # exit 1
```
Comment (bitdivine) [inline .github/workflows/deploy-to-environment.yml:48]: Disabled for testing.  I actually believe that it should be possible to push any ref, not just head of main.  E.g. it should be possible to deploy a tagged release candidate to beta even if the tag is no longer at head of main.  But that is the subject of another PR.

File: scripts/deploy.backend.sh [hunk 2]
```diff
@@ -60,7 +60,7 @@ if [ -n "${ENV+1}" ]; then
          };
          ic_root_key_der = $ic_root_key_der;
      }
-  })" --network "$ENV"
+  })" --network "$ENV" "${@}"
```
Comment (bitdivine) [inline scripts/deploy.backend.sh:63]: This passes cli flags through to `dfx`.

File: .github/workflows/deploy-to-environment.yml [hunk 3]
```diff
@@ -154,14 +154,17 @@ jobs:
 
       - name: Deploy Backend to Environment
         run: |
+          DFX_DEPLOY_FLAGS=() # Arguments to be added to the dfx deploy command.
+          [[ "${{ github.event.inputs.force-backend }}" != "true" ]] || DFX_DEPLOY_FLAGS+=('--yes' '--upgrade-unchanged') # Corresponds to: dfx deploy --yes --upgrade-unchanged
```
Comment (peterpeterparker) [inline .github/workflows/deploy-to-environment.yml:158]: `!= "true"` is correct here? it's not the contrary?

Comment (bitdivine) [reply .github/workflows/deploy-to-environment.yml:158]: It is correct.  If true, more flags are added.  If undefined, false or any other value, no flags are added.

Comment (peterpeterparker) [reply .github/workflows/deploy-to-environment.yml:158]: Gotcha.

**Review by peterpeterparker — APPROVED** [review]
LGTM, thx

**Pull request merged**